src/allmydata/node.py

```diff
@@ -273,12 +312,12 @@ def init_tempdir(self):
         # tempfile.TemporaryFile) to put large request bodies in the given
         # directory. Without this, the default temp dir is usually /tmp/,
         # which is frequently too small.
-        test_name = tempfile.mktemp()
+        self.tempfile_fd, test_name = tempfile.mkstemp()
```

Comment:
It seems like the only change in addition to switching to `tempfile.mkstemp()` that is needed to make this work is `os.close(tempfile_fd)`.  This temporary file doesn't seem to be _used_ anywhere beyond the next line so there's no reason to keep the descriptor open.  It seems the intention is only to verify that temporary files get created in the right directory.  If you revert the context manager related changes and just `os.close(tempfile_fd)` here I would expect things to be in good shape.

Reply:
I had a similar thought, but wasn't sure. Thank you for helping and pushing.